src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/SnippetResponseRewriter.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Completion.Delegation;
+
+/// <summary>
+/// Modifies delegated snippet completion items
+/// </summary>
+/// <remarks>
+/// At the moment primarily used to modify C# "using" snippet to "using statement" snippet
+/// in order to disambiguate it from Razor "using directive" snippet
+/// </remarks>
+internal class SnippetResponseRewriter : DelegatedCompletionResponseRewriter
+{
+    private static readonly IReadOnlyDictionary<string, string> s_snippetToLabel = new Dictionary<string, string>()
```

Comment:
Nit: Just store this as a `Dictionary`. No point pretending with an interface. If you really want readonly, then ImmutableDictionary or FrozenDictionary.